analysis_engine/td/extract_df_from_redis.py

```diff
@@ -44,46 +44,193 @@ def extract_option_calls_dataset(
+    status = ae_consts.NOT_RUN
+    try:
+        redis_rec = redis_get.get_data_from_redis_key(
+            label=label,
+            host=redis_host,
+            port=redis_port,
+            db=redis_db,
+            password=work_dict.get('password', None),
+            key=redis_key,
+            decompress_df=True)
 
-    :param work_dict: dictionary of args
-    :param scrub_mode: type of scrubbing handler to run
-    """
-    return extract_option_dataset('puts', work_dict, scrub_mode=scrub_mode)
-# end of extract_option_puts_dataset
+        status = redis_rec['status']
+        if verbose:
+            log.info(
+                f'{label} - {df_str} redis get data key={redis_key} '
+                f'status={ae_consts.get_status(status=status)}')
 
+        if status == ae_consts.SUCCESS:
+            calls_json = None
+            if 'calls' in redis_rec['rec']['data']:
+                calls_json = redis_rec['rec']['data']['calls']
+            else:
+                calls_json = redis_rec['rec']['data']
+            if verbose:
+                log.info(f'{label} - {df_str} redis convert calls to df')
+            exp_date_str = None
+            try:
+                calls_df = pd.read_json(
+                    calls_json,
+                    orient='records')
+                if len(calls_df.index) == 0:
+                    return ae_consts.SUCCESS, None
+                if 'date' not in calls_df:
+                    log.debug(
+                        'failed to find date column in TD calls '
+                        f'df={calls_df} from lens={len(calls_df.index)}')
+                    return ae_consts.SUCCESS, None
+                calls_df.sort_values(
+                        by=[
+                            'date',
+                            'strike'
+                        ])
+                """
+                for i, r in calls_df.iterrows():
+                    print(r['date'])
+                convert_epochs = [
+                    'ask_date',
+                    'bid_date',
+                    'trade_date'
+                ]
+                for c in convert_epochs:
+                    if c in calls_df:
+                        calls_df[c] = pd.DatetimeIndex(pd.to_datetime(
+                            calls_df[c],
+                            format=ae_consts.COMMON_TICK_DATE_FORMAT
+                        )).tz_localize(
+                            'UTC').tz_convert(
+                                'US/Eastern')
+                # dates converted
+                """
+                exp_date_str = (
+                    calls_df['exp_date'].iloc[-1])
+
+                calls_df['date'] = calls_df['date'].dt.strftime(
+                    ae_consts.COMMON_TICK_DATE_FORMAT)
+
+            except Exception as f:
+                not_fixed = True
+                if (
+                        f'Can only use .dt accessor with '
+                        f'datetimelike values') in str(f):
```

Comment:
"datetimelike" or "datetime like"?

Reply:
this is a pandas error and only happens for data collected prior to the migration done back in jan. if you hit the data must be corrected by backing up to aws, deleting the data in redis, then restoring from the backup. the restore was fixed a while back but this was left so i can remember how it worked if others hit it